src/layout/msWorkspace/msWindow.js

```diff
@@ -321,7 +322,7 @@ var MsWindow = GObject.registerClass(
 
             // If there is no need to maximize, unmaximize, resize or move discard
             if (
-                (shouldBeMaximized && isMaximized) ||
+                (shouldBeMaximized && isMaximized && !needToMoveOrResize) ||
```

Comment:
Is there a reason to add this? The difference between maximize and minimize was that maximize was also the final size and didn't need any resize after that while minimize was just a step and needed to be resize too.

Reply:
Nope. I just tend to code more explicitly to make the code more readable, even if it is a little bit redundant.